kie-ci/src/main/java/org/kie/scanner/KieInJarModuleMetaDataImpl.java

```diff
@@ -136,7 +136,8 @@ private Class<?> getClass(String name) throws ClassNotFoundException {
 
         private byte[] loadClassData(String name) throws IOException {
             for (URL url : urls) {
-                URL tryUrl = new URL(url.toString() + "!/" + name);
+                String prefix = "file".equals(url.getProtocol()) ? "jar:" : "";
```

Comment:
In case of url being null, getProtocol will throw a NPE, do you think we should protect against it?
```suggestion
                String prefix = Objects.equals("file", url.getProtocol()) ? "jar:" : "";
```

Reply:
Hi @gmunozfe , `Objects.equals("file", url.getProtocol())` still throws NPE when url is null. Shall I add an explicit null check?
```java
            for (URL url : urls) {
                if (url == null) {
                    continue;
                }
                String prefix = "file".equals(url.getProtocol()) ? "jar:" : "";
```

Reply:
Sure, I think it's safer also for toString operations later